### PR TITLE
Test Image distributor: Consider TestImageImports from all clusters

### PR DESCRIPTION
**Pull request by alvaroaleman**
Based on top of https://github.com/openshift/ci-tools/pull/2006, please review that one first and only the second commit here.

Makes the TestImageDistributor react to testImageStreamImports in all clusters rather than just the ones in app.ci

Ref https://issues.redhat.com/browse/DPTP-1640

### CODE REVIEW

File: pkg/api/testimagestreamtagimport/v1/types.go [hunk 1]
```diff
@@ -31,12 +31,13 @@ const (
 
 // SetImageStreamLabels sets namespace and name labels so we can easily
 // filter for specific imports
-func (t *TestImageStreamTagImport) SetImageStreamLabels() {
+func (t *TestImageStreamTagImport) SetImageStreamLabels() *TestImageStreamTagImport {
```
Comment (stevekuznetsov) [inline pkg/api/testimagestreamtagimport/v1/types.go:34]: While I see what you did in the test code with this, a setter that returns a reference to the object itself is definitely surprising code ...

File: pkg/controller/test-images-distributor/test_images_distributor.go [hunk 2]
```diff
@@ -74,6 +74,19 @@ func AddToManager(mgr manager.Manager,
 		}
 		buildClusters.Insert(buildClusterName)
 		r.buildClusterClients[buildClusterName] = imagestreamtagwrapper.MustNew(buildClusterManager.GetClient(), buildClusterManager.GetCache())
+
+		if buildClusterName == "app.ci" {
```
Comment (stevekuznetsov) [inline pkg/controller/test-images-distributor/test_images_distributor.go:78]: Do we not have a constant for this?

File: pkg/controller/test-images-distributor/test_images_distributor.go [hunk 3]
```diff
@@ -108,6 +121,20 @@ func AddToManager(mgr manager.Manager,
 	return nil
 }
 
+func testImageStreamTagImportHandlerForNamedCluster(clusterName string) handler.EventHandler {
+	return handler.EnqueueRequestsFromMapFunc(func(o ctrlruntimeclient.Object) []reconcile.Request {
+		testimagestreamtagimport, ok := o.(*testimagestreamtagimportv1.TestImageStreamTagImport)
+		if !ok {
+			logrus.WithField("type", fmt.Sprintf("%T", o)).Error("Got object that was not an ImageStram")
```
Comment (stevekuznetsov) [inline pkg/controller/test-images-distributor/test_images_distributor.go:128]: `s/ImageStram/ImageStream/`

File: pkg/api/testimagestreamtagimport/v1/types.go (hunk 1 above)
Comment (alvaroaleman) [reply pkg/api/testimagestreamtagimport/v1/types.go]: I've renamed it to `WithImageStreamLabels` to clarify the semantics, WDYT?

Comment (alvaroaleman) [reply pkg/api/testimagestreamtagimport/v1/types.go]: FWIW the `sets` from apimachinery also return themselves in their setter: https://github.com/openshift/ci-tools/blob/fa75e5c8447f70771bdf3346b6cfc0bec9875218/vendor/k8s.io/apimachinery/pkg/util/sets/string.go#L49-L54

Comment (stevekuznetsov) [reply pkg/api/testimagestreamtagimport/v1/types.go]: Wow, I had no idea.

**Pull request merged**